packages/frontend/src/components/common/Table/ScrollableTable/TableBody.tsx

```diff
@@ -79,34 +81,89 @@ const TableRow: FC<TableRowProps> = ({
                     getConditionalRuleLabel,
                 );
 
+                const toggleExpander = row.getToggleExpandedHandler();
+                const fontColor =
+                    conditionalFormattingColor &&
+                    readableColor(conditionalFormattingColor) === 'white'
+                        ? 'white'
+                        : undefined;
+
                 return (
                     <BodyCell
                         minimal={minimal}
                         key={cell.id}
                         style={meta?.style}
                         backgroundColor={conditionalFormattingColor}
-                        fontColor={
-                            conditionalFormattingColor &&
-                            readableColor(conditionalFormattingColor) ===
-                                'white'
-                                ? 'white'
-                                : undefined
-                        }
+                        fontColor={fontColor}
                         className={meta?.className}
                         index={index}
                         cell={cell}
                         isNumericItem={isNumericItem(meta?.item)}
                         hasData={!!meta?.item}
                         cellContextMenu={cellContextMenu}
                         isLargeText={
-                            (cellValue?.value.formatted || '').length >
+                            (cellValue?.value?.formatted || '').length >
                             SMALL_TEXT_LENGTH
                         }
                         tooltipContent={tooltipContent}
                     >
-                        {flexRender(
-                            cell.column.columnDef.cell,
-                            cell.getContext(),
+                        {cell.getIsGrouped() ? (
+                            <>
+                                <button
+                                    {...{
+                                        onClick: (e) => {
+                                            e.stopPropagation();
+                                            e.preventDefault();
+                                            toggleExpander();
+                                        },
+                                        style: {
+                                            cursor: row.getCanExpand()
+                                                ? 'pointer'
+                                                : 'normal',
+                                            border: 'none',
+                                            background: 'none',
+                                            float: 'left',
+                                        },
+                                    }}
```

Comment:
I think an extra spread here is not a requirement

Reply:
Nice catch!